grassroots-backend/src/auth/Auth.controller.ts

```diff
@@ -9,83 +9,87 @@ import {
 } from "@nestjs/common";
 import { Response as ExpressResponse } from "express";
 import type { GrassrootsRequest } from "../types/GrassrootsRequest";
-import { ConfigService } from "@nestjs/config";
 import { LoginStateDTO } from "../grassroots-shared/LoginState.dto";
 import { VoidDTO } from "../grassroots-shared/Void.dto";
-import { ApiProperty, ApiQuery, ApiResponse } from "@nestjs/swagger";
 import { PublicRoute } from "./PublicRoute.decorator";
 import { OAuthGuard } from "./OAuth.guard";
+import { getEnvironmentVariables } from "../GetEnvironmentVariables";
 
 @Controller("auth")
 export class AuthController {
-  constructor(private configService: ConfigService) {}
+  // Removed caching as Tim suggested - getEnvironmentVariables() already has cache
 
   // The frontend can redirect here to trigger login.
   @Get("login")
   @UseGuards(OAuthGuard)
   @PublicRoute()
-  @ApiQuery({ name: "redirect_path", type: String })
-  login(@Query() redirect_path: string): VoidDTO {
+  login(@Query("redirect_path") redirectPath: string): VoidDTO {
     // The redirect path is used by the OAuth guard.
-    void redirect_path;
+    void redirectPath;
     return VoidDTO.get();
   }
 
   @Get("google/callback")
   @UseGuards(OAuthGuard)
   @PublicRoute()
-  @ApiProperty()
-  googleAuthRedirect(
+  async googleAuthRedirect(
     @Request() req: GrassrootsRequest,
     @Response() response: ExpressResponse,
-  ): VoidDTO {
-    const host = this.configService.get<string>("FRONTEND_HOST");
+  ): Promise<VoidDTO> {
+    // Call getEnvironmentVariables directly each time (no caching)
+    const environmentVariables = await getEnvironmentVariables();
+
+    const host = environmentVariables.FRONTEND_HOST;
     if (host === undefined) {
       throw new Error("Missing env variable for FRONTEND_HOST");
     }
-    if (!req.user) {
+    if (req.user === undefined) {
```

Comment:
I'm stlil not sure why the eslint rule wasn't catching this for me locally.

Regardless, thanks for fixing it.